AppCenterAnalytics/AppCenterAnalyticsTests/MSEventPropertiesTests.m

```diff
@@ -0,0 +1,187 @@
+#import "MSBooleanTypedProperty.h"
+#import "MSDateTimeTypedProperty.h"
+#import "MSDoubleTypedProperty.h"
+#import "MSEventProperties.h"
+#import "MSEventPropertiesInternal.h"
+#import "MSLongTypedProperty.h"
+#import "MSStringTypedProperty.h"
+#import "MSTestFrameworks.h"
+#import "MSUtility+Date.h"
+
+@interface MSEventPropertiesTests : XCTestCase
+
+@end
+
+@implementation MSEventPropertiesTests
+
+- (void)testSetBoolForKey {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+  BOOL value = YES;
+  NSString *key = @"key";
+
+  // When
+  [sut setBool:value forKey:key];
+
+  // Then
+  MSBooleanTypedProperty *property = (MSBooleanTypedProperty*)sut.properties[key];
+  XCTAssertEqual(property.name, key);
+  XCTAssertEqual(property.value, value);
+}
+
+- (void)testSetInt64ForKey {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+  int64_t value = 10;
+  NSString *key = @"key";
+
+  // When
+  [sut setInt64:value forKey:key];
+
+  // Then
+  MSLongTypedProperty *property = (MSLongTypedProperty*)sut.properties[key];
+  XCTAssertEqual(property.name, key);
+  XCTAssertEqual(property.value, value);
+}
+
+- (void)testSetDoubleForKey {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+  double value = 10.43e3;
+  NSString *key = @"key";
+
+  // When
+  [sut setDouble:value forKey:key];
+
+  // Then
+  MSDoubleTypedProperty *property = (MSDoubleTypedProperty*)sut.properties[key];
+  XCTAssertEqual(property.name, key);
+  XCTAssertEqual(property.value, value);
+}
+
+- (void)testSetDoubleForKeyWhenValueIsInfinity {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+
+  // When
+  [sut setDouble:INFINITY forKey:@"key"];
+
+  // Then
+  XCTAssertEqual([sut.properties count], 0);
+}
+
+- (void)testSetDoubleForKeyWhenValueIsNaN {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+
+  // When
+  [sut setDouble:NAN forKey:@"key"];
+
+  // Then
+  XCTAssertEqual([sut.properties count], 0);
+}
+
+- (void)testSetStringForKey {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+  NSString *value = @"value";
+  NSString *key = @"key";
+
+  // When
+  [sut setString:value forKey:key];
+
+  // Then
+  MSStringTypedProperty *property = (MSStringTypedProperty*)sut.properties[key];
+  XCTAssertEqual(property.name, key);
+  XCTAssertEqual(property.value, value);
+}
+
+- (void)testSetDateForKey {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+  NSDate *value = [NSDate new];
+  NSString *key = @"key";
+
+  // When
+  [sut setDate:value forKey:key];
+
+  // Then
+  MSDateTimeTypedProperty *property = (MSDateTimeTypedProperty*)sut.properties[key];
+  XCTAssertEqual(property.name, key);
+  XCTAssertEqual(property.value, value);
+}
+
+- (void)testSerializeToArray {
+
+  // If
+  MSEventProperties *sut = [MSEventProperties new];
+  MSTypedProperty *property = OCMPartialMock([MSTypedProperty new]);
```

Comment:
I'd rather leave as is. If a child class is failing to do something, then we should see that failure in the test for that type. But if this test fails, it should be solely because event properties is doing something wrong.

Reply:
We talked about this offline. The comment was just my opinion and inconsistent with below test but decided to keep it as it is. Should be okay.